install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -1925,6 +1925,15 @@ envoy:
           labelSelector:
             matchLabels:
               k8s-app: cilium-envoy
+    podAffinity:
```

Comment:
Why is this in the values and not the daemonset? I would rather have this be in the chart, and then overridable if provided.

Reply:
good point.

Updated to actually enforce cilium-envoy to always have pod-affinity with cilium-agent, as cilium-envoy pods will be just no use without agent.

Reply:
Thinking about this again, it's hard to compose different attributes under affinity (e.g. affinity.{podAffinity,podAntiAffinity}.{requiredDuringSchedulingIgnoredDuringExecution,preferredDuringSchedulingIgnoredDuringExecution} in a clean way. Hence, I think we can just keep default value in values.yaml as what it is now. Another benefit is consistency across multiple components.